File: search-application/src/main/java/com/example/search/application/savedsearch/port/in/SavedSearchQuotaExceededException.java
package com.example.search.application.savedsearch.port.in;

import com.example.search.domain.savedsearch.SavedSearch;

/**
 * 한 사용자의 SavedSearch 등록 상한 ({@link SavedSearch#MAX_PER_OWNER}) 초과.
 */
public class SavedSearchQuotaExceededException extends RuntimeException {

    private static final long serialVersionUID = 1L;

    public SavedSearchQuotaExceededException(String ownerId, int currentCount) {
        super("SavedSearch quota 초과 owner=%s current=%d max=%d"
                .formatted(ownerId, currentCount, SavedSearch.MAX_PER_OWNER));
    }
}

File: search-application/src/main/kotlin/com/example/search/application/savedsearch/port/in/SavedSearchQuotaExceededException.kt
package com.example.search.application.savedsearch.port.`in`

import com.example.search.domain.savedsearch.SavedSearch

/**
 * 한 사용자의 SavedSearch 등록 상한 ([SavedSearch.MAX_PER_OWNER]) 초과.
 */
class SavedSearchQuotaExceededException(ownerId: String, currentCount: Int) : RuntimeException(
    "SavedSearch quota 초과 owner=$ownerId current=$currentCount max=${SavedSearch.MAX_PER_OWNER}"
) {
    companion object {
        private const val serialVersionUID: Long = 1L
    }
}
